apps/backend/src/user/user.service.ts

```diff
@@ -101,4 +105,46 @@ export class UserService {
       throw new NotFoundException(`User with id ${id} not found`);
     }
   }
+
+  async saveProfilePicture(authSchId: string, buffer: Buffer, mimetype: string) {
+    if (mimetype !== 'image/png' && mimetype !== 'image/jpeg') {
+      throw new BadRequestException('Invalid image format');
+    }
+    try {
+      await this.createOrUpdateProfilePicture(authSchId, buffer);
+    } catch (error) {
+      throw new NotFoundException(`User with id ${authSchId} not found`);
+    }
+  }
+
+  async findProfilePicture(authSchId: string): Promise<Buffer> {
+    try {
+      const profilePic = await this.prisma.profilePicture.findUniqueOrThrow({ where: { userId: authSchId } });
+      return profilePic.profileImage;
+    } catch (error) {
+      throw new NotFoundException(`User with id ${authSchId} not found`);
+    }
+  }
+
+  private async createOrUpdateProfilePicture(userId: string, profileImage: Buffer) {
+    const { image, mimeType } = await this.optimizeImage(profileImage);
+    const data = { userId, profileImage: image, mimeType };
+    try {
+      await this.prisma.profilePicture.update({ where: { userId: userId }, data });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === 'P2025') {
+          await this.prisma.profilePicture.create({ data });
```

Comment:
the first upload wasn't working but I didn't realise why, finally found it. You have to return the result of this line, otherwise the error will be thrown after the user is updated

Reply:
omg, good catch